src/LocalNeuralNet.cpp

```diff
@@ -0,0 +1,271 @@
+    {
+        m_vec.resize(dim);
+    }
+
+    std::size_t LocalNeuralNetImp::TensorOneD::get_dim() const
+    {
+        return m_vec.size();
+    }
+
+    LocalNeuralNetImp::TensorOneD::TensorOneD(std::vector<float> input)
+    {
+        set_dim(input.size());
+        std::size_t vec_size = m_vec.size();
+
+        for (std::size_t idx = 0; idx < vec_size; ++idx) {
+            m_vec[idx] = input[idx];
+        }
+    }
+
+    LocalNeuralNetImp::TensorOneD LocalNeuralNetImp::TensorOneD::operator+(const LocalNeuralNetImp::TensorOneD& other)
+    {
+        if (get_dim() != other.get_dim()) {
+            throw geopm::Exception("Adding vectors of mismatched dimensions.",
+                                   GEOPM_ERROR_INVALID, __FILE__, __LINE__);
+        }
+
+        LocalNeuralNetImp::TensorOneD rval(m_vec.size());
+        std::transform(m_vec.begin(), m_vec.end(), other.m_vec.begin(), rval.m_vec.begin(), std::plus<float>());
+
+        return rval;
+    }
+
+    LocalNeuralNetImp::TensorOneD LocalNeuralNetImp::TensorOneD::operator-(const LocalNeuralNetImp::TensorOneD& other)
+    {
+        if (get_dim() != other.get_dim()) {
+            throw geopm::Exception("Subtracting vectors of mismatched dimensions.",
+                                   GEOPM_ERROR_INVALID, __FILE__, __LINE__);
+        }
+
+        LocalNeuralNetImp::TensorOneD rval(m_vec.size());
+        std::transform(m_vec.begin(), m_vec.end(), other.m_vec.begin(), rval.m_vec.begin(), std::minus<float>());
+
+        return rval;
+    }
+
+
+    float LocalNeuralNetImp::TensorOneD::operator*(const LocalNeuralNetImp::TensorOneD& other)
+    {
+        if (get_dim() != other.get_dim()) {
+            throw geopm::Exception("Inner product of vectors of mismatched dimensions.",
+                                   GEOPM_ERROR_INVALID, __FILE__, __LINE__);
+        }
+
+        return std::inner_product(m_vec.begin(), m_vec.end(), other.m_vec.begin(), 0);
+    }
+
+    LocalNeuralNetImp::TensorOneD& LocalNeuralNetImp::TensorOneD::operator=(const LocalNeuralNetImp::TensorOneD &other)
+    {
+        m_vec = other.m_vec;
+        return *this;
+    }
+
+    LocalNeuralNetImp::TensorOneD& LocalNeuralNetImp::TensorOneD::operator=(LocalNeuralNetImp::TensorOneD &&other)
+    {
+        if (&other == this) {
+            return *this;
+        }
+
+        m_vec = std::move(other.m_vec);
+        return *this;
+    }
+
+    float& LocalNeuralNetImp::TensorOneD::operator[] (std::size_t idx)
+    {
+        return m_vec.at(idx);
+    }
+
+    float LocalNeuralNetImp::TensorOneD::operator[] (std::size_t idx) const
+    {
+        return m_vec.at(idx);
+    }
+
+    LocalNeuralNetImp::TensorOneD LocalNeuralNetImp::TensorOneD::sigmoid() const
+    {
+        std::size_t vec_size = m_vec.size();
+        LocalNeuralNetImp::TensorOneD rval(vec_size);
+        for(std::size_t idx = 0; idx < vec_size; idx++) {
+            rval[idx] = 1/(1 + expf(-(m_vec.at(idx))));
```

Comment:
Nit: you've ensured `0 < vec_size < m_vec.size()`, so you could simply use the indexing operator.